rank/pages/search.tsx

```diff
@@ -1,59 +1,70 @@
+import { Env, Index, Namespace } from '../types/searchTemplate'
 import { GetServerSideProps, NextPage } from 'next'
 
-import { Env } from '../types/env'
 import Hit from '../components/Hit'
-import { Namespace } from '../types/namespace'
 import QueryForm from '../components/QueryForm'
-import { ApiResponse as SearchApiResponse } from './api/search'
+import { SearchResponse } from '../types/elasticsearch'
 import absoluteUrl from 'next-absolute-url'
+import { listIndices } from '../services/search-templates'
 
 type Props = {
-  data: SearchApiResponse
+  data: SearchResponse
   search: {
     query?: string
     namespace?: Namespace
+    index?: Index
     env: Env
   }
+  indices: Index[]
 }
 
 export const getServerSideProps: GetServerSideProps<Props> = async ({
   query: qs,
   req,
 }) => {
-  const query = qs.query ? qs.query.toString() : undefined
-  const namespace = qs.namespace ? qs.namespace.toString() : undefined
-  const env = qs.env ? qs.env.toString() : undefined
-  const { origin } = absoluteUrl(req)
-  const reqQs = Object.entries({ query, namespace, env })
-    .filter(([, v]) => Boolean(v))
-    .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
-    .join('&')
+  const indices = await listIndices()
 
-  const data: SearchApiResponse = await fetch(
-    `${origin}/api/search?${reqQs}`
-  ).then((res) => res.json())
+  const query = qs.query ? qs.query.toString() : ''
+  const env = qs.env ? qs.env.toString() : 'local'
+  const index = qs.env ? qs.index.toString() : indices[0]
+
+  let data: SearchResponse = {
```

Comment:
This looks a little untidy. What are we trying to say here? It looks a little like "If there is not query, there is no data" so why not just make data null?

Reply:
totally right!